torch/csrc/distributed/rpc/py_rref.cpp

```diff
@@ -61,13 +61,13 @@ py::object PyRRef::localValue() {
     // grabbing the GIL.
     pybind11::gil_scoped_acquire ag;
     res = torch::jit::toPyObject(std::move(value));
+    PythonRpcHandler::getInstance().handleExceptionGILHeld(res);
```

Comment:
@jjlilley it could rarely happen for some cases, this thread one is waiting for PythonRpcHandler to be constructed by thread two,  thread two needs to acquire the GIL to construct the PythonRpcHandler, but thread two can not get the GIL, because thread one is holding it in this scope. In this rare case, dead lock could happen though

Reply:
Ah, thanks for the clarification. 
but it seems like I could safely acquire the PythonRpcHandler outside the lock, right?

  auto &rpcHandler = PythonRpcHandler::getInstance();
  pybind11::gil_scoped_acquire ag;
  res = torch::jit::toPyObject(std::move(value));
  rpcHandler.handleExceptionGILHeld(res);

Reply:
yes, this works, thanks!